src/python/WMComponent/TaskArchiver/TaskArchiverPoller.py

```diff
@@ -690,6 +689,16 @@ def archiveWorkflowSummary(self, spec):
             workflowData['output'][dataset]['events'] = entry['events']
             workflowData['output'][dataset]['tasks']  = outputList.get(dataset, {}).keys()
 
+        # If the workflow was aborted, then don't parse all the jobs, cut at 5k
+        try:
+            reqDetails = self.centralCouchDBWriter.getRequestByNames(workflowName)
+            wfStatus = reqDetails[workflowName]['RequestTransition'][-1]['Status']
+            if wfStatus in ["aborted", "aborted-completed", "aborted-archived"]:
+                logging.info("Workflow %s in status %s with a total of %d jobs, capping at 5000" % (
+                    workflowName, wfStatus, len(failedJobs)))
+                failedJobs = failedJobs[:5000]
```

Comment:
Should we need to keep 5000 data? I am just wondering what is the right number. How many failed job was there when this was causing problem?